airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/csv/NoFlatteningSheetGenerator.java

```diff
@@ -21,6 +22,11 @@ public List<String> getHeaderRow() {
         JavaBaseConstants.COLUMN_NAME_DATA);
   }
 
+  @Override
```

Comment:
this can be stubbed out in `BaseSheetGenerator` so this doesn't leak into this class or the RootLevelFlattening class.

Reply:
it looks like all of these classes will benefit from the same pattern we are doing here where we only pass in the data and the emitted at timestamp.

again, can do in a follow up PR to avoid boiling the ocean. I would leave a todo so this is not a dangling comment.